SalesforceCommerceCloud/pwa-kit
# [Localization] Fix the pseudo locale regression

# Description

Ticket: https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00000DHiZqYAL/view
Summary of bug: enabling pseudo locale would result in a server error, due to accidentally sending `en-XB` to the commerce API.

First of all, there's already this [small PR](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/174) that actually fixed the pseudo locale regression. But upon closer look at 'commerce-api/index.js', there were other subtle bugs to be found:
- If `sendLocale = false`, we were still sending out the locale
  - It turns out that when destructuring the `parameters` object, we didn't delete properly.
- All parts of the app, except for our commerce-api client, were aware of the pseudo locale
  - In this PR, the pseudo locale is now isolated to only the dynamic import part. This has the upside of no longer updating the page urls with the pseudo locale.

# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- Fixed the deletion of `parameters` object properties
- The only part of our source code that is aware of the pseudo locale is now isolated to just `loadLocaleData()`
- Some refactoring

# How to Test-Drive This PR

1. `npm run start:pseudolocale`
2. Go to http://localhost:3000/
3. Verify that the homepage is using the pseudo locale. You may have to reload the page if the browser was caching it.
3. Tap on the button "[!! Ṡḫŏŏŏṗ ŊŊŊèẘ Ḁŕŕŕŕíṿâââĺś !!]"
4. Change the locale to French
5. Verify that the page url is correct, and that it's not using the pseudo locale "en-XB"

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
